model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -348,7 +349,9 @@ class Diffusion:
     """Class that configures diffusion and does one diffusion step."""
 
     def __init__(
```

Comment:
There is the `self._allocate_local_fields` function that  that should also use the backend. It uses `gt4py_field_allocation` functions which in turn call the `gtx.as_field` which can take a `allocator = backend.allocator` argument. So you might have to pass it through the functions in `gt4py_field_allocation`

Reply:
If you don't set the self._backend you will have to to do call the `_allocate_temporary_fields` in the constructor. (which we want to do anyway as we want to do everything that we do in the `def init` to the constructor. (see below)